creme/replay/base.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+
+import collections
+import random
+
+from .. import base
+
+
+class ReplayBuffer:
+    """ReplayBuffer
+
+    ReplayBuffer stores a dictionnary of tuples  (features, target). The tuples stored in the
+    buffer are the observations which are difficult for a model to regress or classify.The buffer is
+    used to re-train a model on targets that are difficult to extrapolate. When fitting the model,
+    it is re-trained with probability p on one of the observations (sampled uniformly) in the
+    buffer. The model is re-trained with a probability (1 - p) on the new data.
+
+    To update the buffer, you must specify the loss of a given tuple (features, target). This new
+    observation will be stored if the loss is sufficiently high compared to the losses already
+    stored in the buffer.
+
+    Parameters:
+        size (int): Number of stored tuples (features, target).
+        p (float): Probability to update the model with an observation from the buffer when fitting
+            on a new observation. 0. <= p <= 1.
+        seed (int): Random state.
+
+    """
+    def __init__(self, size, p, seed=None):
+        self.size = size
+        self.p = p
+        self.buffer = {}
+        # Intiliaze the losses history with respect to the buffer size with losses equals to -1.
+        self.loss_history = {key: -1. for key in range(self.size)}
+        # Initialize the minimum loss observed to -1.
+        self.min_loss = -1.
+        # Initialize the key associated to the minimum loss observed.
+        self.key_min_loss = 0
+        self.seed = seed
+        self._rng = np.random.RandomState(seed)
```

Comment:
```
from sklearn import utils
self.random_state = utils.check_random_state(random_state)
```

Reply:
Actually if possible I think it's preferable to use the standard library instead of depending on `sklearn` :).

Reply:
Should we create our own `utils.check_random_state` and remove sklearn other dependencies?

Reply:
Using `self._rng = random.Random(seed)` works just fine. I prefer to `seed` to `random_state` namewise. We don't need any utility function!